No [initial] speech feedback when using Windows built-in text to speech

### Is there an existing issue for this?

- [x] This is a bug in RetroArch frontend
- [x] I have searched the existing issues

### Description

Using portable copies of Retroarch 1.20 or 1.19.1 on Windows 11 24H2:
after enabling accessibility mode by changing the option to "true" in the retroarch.cfg file, there is no initial speech feedback when launching retroarch; the initial "retroarch accessibility mode on, main menu" is not spoken, and nothing is spoken if arrow keys are pressed to navigate the menus. However, if I press ESC to quit then retroarch does speak the "press again to quit" phrase using the Windows TTS voice; at this point, if arrow keys are press instead of pressing ESC again, retroarch will continue to speak menus and options as it did in 1.17, however there is a slight delay to the speech, about 0.75 to 1.0 second. So every time retroarch is launched, I have to press ESC, wait for the "press again to quit" phrase to be spoken, then the arrow keys can be used to navigate the interface with speech. This happens whether  I run retroarch with no screen reader loaded or if I run retroarch with either Narrator or JAWS screen readers running.
I started using RA a few years ago, and it always used the Windows TTS to speak, so I was unaware that it actually has some kind of special support for the NVDA screen reader, until I read the log file I created for this report. So I then tried retroarch while running the NVDA screen reader and it does work as expected and gives speech feedback using NVDA's speech synthesizer and settings; it does so at an amazing speed too, meaning there's no noticeable delay from when a key is pressed and the menu option is spoken!
Is using RA with NVDA the new way to get accessibility working going forward? Or will Windows text to speech also continue to be supported? If both will be supported, what is the logic RA uses to automatically determine which speech system/TTS to  use? From reading the log it seems that RA is initially looking for the NVDA screen reader, which it obviously can't find when I don't have it running. If RA is to work with both Windows TTS and NVDA, should this (which speech system to use) be an added option in the accessibility settings? I don't mind using the NVDA option if that is what works, but then I realized the devs might not be aware that Windows TTS is broken; so I decided to file the bug report, in case they intend to also keep it as a speech option.
A final question which I'd be happy to add as a request if need-be, could such integration (speaking through the screen reader's synthesizer) also be added for the JAWS screen reader? It would be nice to have for those who use JAWS as their primary screen reader, as it would mean not having to unload one screen reader to load another for a quick gaming session :)
In any case, thanks for the work you've put into making RA as accessible as it is thus far, it really does make this front-end a unique piece of software!

[retroarch.log](https://github.com/user-attachments/files/18953528/retroarch.log)

### Expected behavior

Previously, When accessibility mode is enabled, retroarch uses Windows built-in text to speech synthesizer to speak the menu options automatically as they are highlighted. Furthermore, it also speaks an opening announcement when retroarch is launched letting the user know that "accessibility mode is on". This does not occur with either RA 1.19.1 or the official 1.120.0 release; I also tested the latest nightly build as of 02-24-2025, no change. Note that the feature works as expected in release 1.17.

### Steps to reproduce the bug

1. enable accessibility mode by using settings>accessibility or by setting the option to true in retroarch.cfg, make sure no screen readers are running in the background.
2. Close and relaunch retroarch (if applicable)
3. Launch retroarch with accessibility mode enabled, normally it should speak the initial announcement and speak any menu options as they are highlighted, this is not happening on my system.
4. Now press the ESC key, RA will actually speak the "press again to quit" announcement. If you press the arrow keys to highlight menu options, RA will continue to speak them using the Windows built-in TTS as expected.

### Version/Commit

1.20.0

### Bisect Results

1.17.0

### Present in the nightly version

Yes, this is reproduced in the nightly build

### Platform & operating system

Windows 11 24h2

### Affected Cores

n/a

### Environment information

HP ZBook Power G11 laptop, Intel core ultra 7 155h, 32GB RAM

### Relevant log output

```shell
[INFO] RetroArch 1.20.0 (Git 6412f4feef)
[INFO] Spoke: RetroArch accessibility on.  Main Menu Load Core.
[ERROR] Error communicating with NVDA
[INFO] === Build =======================================
[INFO] CPU Model Name: Intel(R) Core(TM) Ultra 7 155H
[INFO] Capabilities: MMX MMXEXT SSE SSE2 SSE3 SSSE3 SSE4 SSE42 AES AVX AVX2 
[INFO] Version: 1.20.0
[INFO] Git: 6412f4feef
[INFO] Built: Jan  4 2025
[INFO] =================================================
[INFO] [Input]: Found input driver: "dinput".
[INFO] [Environ]: SET_PIXEL_FORMAT: RGB565.
[INFO] [Core]: Version of libretro API: 1, Compiled against API: 1
[INFO] [Core]: Geometry: 320x240, Aspect: 1.333, FPS: 60.00, Sample rate: 48000.00 Hz.
[INFO] [Video]: Set video size to: 960x720.
[INFO] [XInput]: Found XInput v1.4.
[INFO] [Joypad]: Found joypad driver: "dinput".
[INFO] [D3D11]: Device created (Feature Level: 11.0)
[INFO] [D3D11]: Flip model and tear control supported and enabled.
[INFO] [D3D11]: Requesting 1 maximum frame latency, using 1.
[INFO] [D3D11]: Found GPU at index 0: "NVIDIA RTX 1000 Ada Generation Laptop GPU".
[INFO] [D3D11]: Found GPU at index 1: "Intel(R) Graphics".
[INFO] [D3D11]: Found GPU at index 2: "Microsoft Basic Render Driver".
[INFO] [D3D11]: Using GPU index 0.
[INFO] [Video]: Found display server: "win32".
[INFO] [Audio]: Set audio input rate to: 48000.00 Hz.
[DEBUG] [WASAPI]: Initializing default eRender device..
[DEBUG] [WASAPI]: Requesting shared 16-bit 2-channel client with WAVE_FORMAT_PCM samples at 48000Hz 64ms.
[DEBUG] [WASAPI]: Default device period is 10.0ms.
[DEBUG] [WASAPI]: Minimum device period is 3.0ms.
[DEBUG] [WASAPI]: Shared stream latency is 0.0ms.
[DEBUG] [WASAPI]: Endpoint buffer size is 3072 frames (6144 samples, 12288 bytes, 64.0 ms).
[INFO] [WASAPI]: Client initialized (shared, PCM, 48000Hz, 64.0ms).
[INFO] [Audio]: Started synchronous audio driver.
[DEBUG] [WASAPI mic]: Initialized microphone driver context.
[INFO] [Microphone]: Initialized microphone driver.
[INFO] [Display]: Found display driver: "d3d11".
[INFO] [MIDI]: Output device: "Microsoft GS Wavetable Synth".
[INFO] [Playlist]: Loading history file: "D:\RetroArch120\content_history.lpl".
[INFO] [Playlist]: Loading history file: "D:\RetroArch120\content_music_history.lpl".
[INFO] [Playlist]: Loading history file: "D:\RetroArch120\content_video_history.lpl".
[INFO] [Playlist]: Loading history file: "D:\RetroArch120\content_image_history.lpl".
[INFO] [Playlist]: Loading favorites file: "D:\RetroArch120\content_favorites.lpl".
[DEBUG] [Audio]: Started audio driver "wasapi" (is_shutdown=false)
[DEBUG] [Audio]: Stopped audio driver "wasapi"
[INFO] Spoke: Press again to quit...
[ERROR] Error communicating with NVDA
[INFO] Spoke: Load Content
[INFO] Spoke: Show Desktop Menu
[INFO] Spoke: Online Updater
[INFO] Spoke: Information
[INFO] Spoke: Configuration File
[INFO] Spoke: Restart 
[INFO] Spoke: Quit 
[INFO] [Config]: Saved new config to "D:\RetroArch120\retroarch.cfg".
[INFO] Spoke: Saved new config to "D:\RetroArch120\retroarch.cfg".
[DEBUG] [Audio]: Started audio driver "wasapi" (is_shutdown=true)
[INFO] Spoke: Main Menu Quit 
[DEBUG] [Audio]: Started audio driver "wasapi" (is_shutdown=true)
[INFO] [Core]: Content ran for a total of: 00 hours, 00 minutes, 00 seconds.
[DEBUG] [Audio]: Stopped audio driver "wasapi"
[INFO] [Core]: Unloading core..
[INFO] [Core]: Unloading core symbols..
[DEBUG] [Video]: Does not have enough samples for monitor refresh rate estimation. Requires to run for at least 4096 frames.
```

--- a/frontend/drivers/platform_win32.c
+++ b/frontend/drivers/platform_win32.c
@@ -1124,9 +1124,12 @@ static bool accessibility_speak_windows(int speed,
       if (!wc || res != 0)
       {
          RARCH_ERR("Error communicating with NVDA\n");
+         /* Fallback on powershell immediately and retry */
+         g_plat_win32_flags &= ~PLAT_WIN32_FLAG_USE_NVDA;
+         g_plat_win32_flags |= PLAT_WIN32_FLAG_USE_POWERSHELL;
          if (wc)
             free(wc);
-         return false;
+         return accessibility_speak_windows(speed, speak_text, priority);
       }
 
       nvdaController_cancelSpeech_func();

